lambda_code/src/models/utils.py

```diff
@@ -0,0 +1,133 @@
+"""Generic utilities for models"""
+
+import re
+from datetime import datetime
+from typing import Literal, Union
+
+
+def generic_string_validation(
+    field_value: str,
+    field_location: str,
+    defined_length: int = 0,
+    max_length: int = 0,
+    predefined_values: tuple = None,
+):
+    """
+    Apply generic validation to a string field to ensure it is a non-empty string which meets
+    the length requirements and predefined values requirements
+    """
+    if not isinstance(field_value, str):
+        raise TypeError(f"{field_location} must be a string")
+
+    if defined_length:
+        if len(field_value) != defined_length:
+            raise ValueError(f"{field_location} must be {defined_length} characters")
+    else:
```

Comment:
Because there is an overarching rule that where no data is available the field must not be included. If the field does not exist then the validator gets passed, so if the validator is being run, and the value is None (we've also interpreted it to mean if the value is some other empty value) then that's never acceptable

Reply:
It's laid out this way because the error message is different for empty strings depending on whether there is a defined length. If there's no defined length then we just wanted the error to say that it must be non-empty, but if there's a defined length it seemed more appropriate to specify what length it should be